server/routes/approvedVisitors.js

```diff
@@ -0,0 +1,57 @@
+const express = require('express');
+
+const createApprovedVisitorsRouter = ({ offenderService }) => {
+  const router = express.Router();
+
+  const getPersonalisation = async (user, rawQuery) => {
+    const { approvedVisitors: rawApprovedVisitors, error = null } =
+      await offenderService.getApprovedVisitorsFor(user);
+    if (error) return { signedInUser: user.getFullName(), error };
+
+    const maxVisitorsPerPage = 10;
+    const totalCount = rawApprovedVisitors.length;
+    const totalPages = Math.ceil(totalCount / maxVisitorsPerPage);
+    const rawPage = Math.round(rawQuery.page) || 1;
+    const page = rawPage > 0 && rawPage <= totalPages ? rawPage : 1;
+    const max = Math.min(page * maxVisitorsPerPage, totalCount);
+    const min = Math.min((page - 1) * maxVisitorsPerPage) + 1;
+    const pageData = { page, totalPages, min, max, totalCount };
```

Comment:
same comment applies from before.. but happy to drop this if you want and we could look at it if we ever have repeat the pattern

> meh.. NBD but I'd have shoved this in a util as the concept of "providing pagination over a list" is pretty generic (non domain specific) concept that could be unit tested in isolation outside of having all the gumpf of testing within a controller